homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,282 @@
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_scale
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._temperature
+
+    @property
+    def current_humidity(self) -> Optional[int]:
+        """Return the current humidity."""
+        return self._humidity
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the hvac mode."""
+        return HVAC_MODES[self._mode]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """List of available hvac modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        target_temperature = None
+        if not self._dual_setpoint:
+            target_temperature = self._target_temperature
+        return target_temperature
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        target_temperature_low = None
+        if self._dual_setpoint:
+            target_temperature_low = self._target_temp_cool
+        return target_temperature_low
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        target_temperature_high = None
+        if self._dual_setpoint:
+            target_temperature_high = self._target_temp_heat
+        return target_temperature_high
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return current preset mode."""
+        return self._setpoint_status
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return preset modes."""
+        return [
+            PRESET_NO_HOLD,
+            PRESET_HOLD_UNTIL,
+            PRESET_TEMPORARY_HOLD,
+            PRESET_PERMANENT_HOLD,
+            PRESET_VACATION_HOLD,
+        ]
+
+    @property
+    def min_temp(self) -> float:
+        """Identify min_temp in Lyric API or defaults if not available."""
+        return self._min_temperature
+
+    @property
+    def max_temp(self) -> float:
+        """Identify max_temp in Lyric API or defaults if not available."""
+        return self._max_temperature
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if self._dual_setpoint:
+            if target_temp_low is not None and target_temp_high is not None:
+                temp = (target_temp_low, target_temp_high)
+        else:
+            temp = kwargs.get(ATTR_TEMPERATURE)
+        _LOGGER.debug("Set temperature: %s", temp)
+        self.device.temperatureSetpoint = temp
```

Comment:
This is doing I/O and so should be done in the executor.

```suggestion
        await self.hass.async_add_executor_job(setattr, self.device, "temperatureSetpoint", temp)
```